src/components/Requirements/RequirementFulfillmentSlots.vue

```diff
@@ -103,16 +104,25 @@ export default defineComponent({
         }
       );
       if (requirementFulfillmentSpec === null) return [];
-      const requirementFulfillmentEligibleCourses = requirementFulfillmentSpec.eligibleCourses;
+      const appliedRequirementFulfillmentSpec =
+        (requirementFulfillmentSpec.additionalRequirements || {})[this.compoundRequirementChoice] ||
+        requirementFulfillmentSpec;
+      const requirementFulfillmentEligibleCourses =
+        appliedRequirementFulfillmentSpec.eligibleCourses;
+      const matchedCourses = (
```

Comment:
Can you add a comment for what `matchedCourses` means? Is it supposed to be the taken courses?